src/main/java/com/yelp/nrtsearch/server/luceneserver/QueryNodeMapper.java

```diff
@@ -58,6 +58,10 @@
               MatchOperator.MUST, BooleanClause.Occur.MUST));
 
   public Query getQuery(com.yelp.nrtsearch.server.grpc.Query query, IndexState state) {
+    if (query.getQueryNodeCase()
+        == com.yelp.nrtsearch.server.grpc.Query.QueryNodeCase.QUERYNODE_NOT_SET) {
```

Comment:
You could probably just add this case to the big switch statement in getQueryNode.

Reply:
Thanks for the suggestion. Moved the logic.